utils/build_swift/legacy_driver_arguments.py

```diff
@@ -0,0 +1,1334 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+"""
+This module is part of a larger migration to a newer argument parsing scheme.
+This legacy argument parser will be removed in the near future!
+"""
+
+from __future__ import print_function
+
+import argparse
+import android.adb.commands
+import multiprocessing
+import platform
+
+from swift_build_support.swift_build_support import arguments
+from swift_build_support.swift_build_support import host
+from swift_build_support.swift_build_support import targets
+from swift_build_support.swift_build_support import workspace
+
+from swift_build_support.swift_build_support.targets import \
+    StdlibDeploymentTarget
+
+
+__all__ = [
+    'create_argparser',
+]
+
+
+class ApplyDefaultsArgumentParser(argparse.ArgumentParser):
```

Comment:
Just to loop back from our conversation, can you do this in a separate commit. I want to be able to eyeball the move of code and then see the refactoring done upon the nicely refactored code. It will make it easier for me to review.

Reply:
Sure thing! I'll chunk out the two large code-moves into separate PRs.

Reply:
Thanks! = )